docs/specs/mlnet-cli/MLNET-CLI-Specs.md

```diff
@@ -0,0 +1,969 @@
+
+The subset of .NET developers who at the same time are skilled on data science and machine learning in general is very small compared to the total number of regular .NET developers.
+
+We need a way to enable regular .NET developers to easily use [ML.NET](https://github.com/dotnet/machinelearning) to create custom models solving typical ML scenarios in the enterprise. 
+
+If we don't provide a really simple way to use [ML.NET](https://github.com/dotnet/machinelearning) for regular developers (almost no data science knowledge at all), then we won't be able to really "democratize" machine learning for .NET developers. 
+
+## Evidence
+
+Sample feedback and quotes:
+
+- *"My blocking issues?: Understanding the pros and cons of various classifiers that are available in libraries and how best to use a  combination. What shoud I use?"*
+
+- *"I'm missing too much ML knowledge. Would be great to know when to pick which learner or if the platform would tell me?"*
+
+- *“It’s confusing to know what tools to use to solve a particular problem.  All of the classes use the language of machine learning experts.”*
+
+- *"lack of hints on the relative benefits of using alternative algorithms”*
+
+- *"Personally - My understanding of certain areas, the samples online are great, however I've found it difficult to deviate from the set pattern (possibly intentional?) which has made it an uphill struggle trying to achieve what I want to"*
+
+- *"Remember that not all of us are trained as statisticians. One example: "FastTree" trainer, we can change "numTrees" and "numLeaves" for example, and the documentation says "Total number of decision trees to create in the ensemble." Okay, but what should I do now ? I have 90'000 data points, should I create 100, 900, 2000 trees ?"*
+
+- *“To be honest, the AI/ML communities do a terrible job of explaining what the heck they're doing to the average developer. For example the Standford lectures that go into GloVe, etc. are talking about taking the derivatives of a formula by lecture 2. Or at the other extreme it's a toy example with no actual explanation of what the heck is going on, just copy this code and run it. Or I have to deal with the fact that every example is done in Python and Jupyter."*
+
+- *"BUT if you want it to really get more use outside of the weird people like me, you need a huge amount of documentation, examples, blog posts, etc. that takes people from 0 knowledge to actual production usage. You need something ELSE to really be a game-changer here for regular developers who are not data scientists"*
+
+# Who is the Customer
+
+- Regular .NET developers getting started with machine learning while trying to use .NET (C# and F# most of all) for ML.
+
+- Specific developer roles are: enterprise developers, start-up developers, ISV developers and internal MSFT teams developers.
+
+
+# Goals
```

Comment:
>Goals [](start = 2, length = 5)

also a goal should be:
- Ease deployment and productization of models via service templates
- Teach best practices via generated code templates.